File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max_profit = 0;
        int mini = prices[0];
        for (int i = 0; i < prices.size(); i++){
            mini = min(mini, prices[i]);
            int cost = prices[i] - mini;
            max_profit = max(max_profit, cost);
        }
        return max_profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        if not prices:
            return 0
            
        max_profit = 0
        mini = prices[0]
        
        
        for price in prices:
            mini = min(mini, price)
            cost = price - mini
            max_profit = max(max_profit, cost)
            
        return max_profit